Atomic/syscontainers.py

```diff
@@ -1317,3 +1493,86 @@ def get_out_checksum(obj): return obj.out_checksum if hasattr(obj, 'out_checksum
+            url = labels.get("url")
+            source0 = labels.get("source0")
+            requires = labels.get("requires")
+            provides = labels.get("provides")
+            conflicts = labels.get("conflicts")
+            description = labels.get("description")
+
+            self._generate_spec_file(spec_file, rpm_content, name, summary, license_, version=version, release=release,
+                                     url=url, source0=source0, requires=requires, provides=provides, conflicts=conflicts,
+                                     description=description, installed_files=installed_files)
+
+            cwd = os.getcwd()
+            cmd = ["rpmbuild", "--noclean", "-bb", spec_file,
+                   "--define", "_sourcedir %s" % temp_dir,
+                   "--define", "_specdir %s" % temp_dir,
+                   "--define", "_builddir %s" % temp_dir,
+                   "--define", "_srcrpmdir %s" % cwd,
+                   "--define", "_rpmdir %s" % cwd,
+                   "--build-in-place",
+                   "--buildroot=%s" % rpm_content]
+            util.write_out(" ".join(cmd))
+            if not self.display:
+                util.check_call(cmd)
+            return False
+        finally:
+            shutil.rmtree(temp_dir)
+
+    def _generate_spec_file(self, out, destdir, name, summary, license_, version="1.0", release="1", url=None,
+                            source0=None, requires=None, conflicts=None, provides=None, description=None, installed_files=None):
+        spec = "%global __requires_exclude_from ^.*$\n"
+        spec = spec + "%global __provides_exclude_from ^.*$\n"
+
+        fields = {"Name" : "%s-%s" % (RPM_NAME_PREFIX, name), "Version" : version, "Release" : release, "Summary" : summary,
+                  "License" : license_, "URL" : url, "Source0" : source0, "Requires" : requires,
+                  "Provides" : provides, "Conflicts" : conflicts}
+        for k, v in fields.items():
+            if v is not None:
+                spec = spec + "%s:\t%s\n" % (k, v)
+
+        spec = spec + "\n%description\n"
+        if description:
+            spec = spec + "%s\n" % description
+
+        spec = spec + "\n%files\n"
+        for root, _, files in os.walk(os.path.join(destdir, "etc")):
+            rel_path = os.path.relpath(root, destdir)
+            for f in files:
+                spec += "%config \"%s\"\n" % os.path.join("/", rel_path, f)
+
+        spec += "/usr/lib/containers/atomic/%s\n" % name
+        for root, _, files in os.walk(os.path.join(destdir, "usr/lib/systemd/system")):
+            for f in files:
+                spec = spec + "/usr/lib/systemd/system/%s\n" % f
+        for root, _, files in os.walk(os.path.join(destdir, "usr/lib/tmpfiles.d")):
+            for f in files:
+                spec = spec + "/usr/lib/tmpfiles.d/%s\n" % f
+        if installed_files:
+            for i in installed_files:
+                spec = spec + "%s\n" % i
+
```

Comment:
Would it make sense to add changelog? So users know that the rpm is generated and installed by `atomic`.